### PR TITLE
Added Basalt + Blackstone to Crucible

### LINKED ISSUE #2369: Adding Basalt to the Crucible and Electrified Crucible
This Issue is part of [Hacktoberfest](https://hacktoberfest.digitalocean.com/) - A yearly event which encourages participation in the open-source community. [Sign up on their website](https://hacktoberfest.digitalocean.com/) and submit **four pull requests during october (Oct 1st - Oct 31st)** to any public open-source project on GitHub and earn a limited-edition T-shirt or plant a tree!

## :mag_right: Scope
The following bullet points explain what the scope of this feature should be.
It should give a general idea of what we want, you can of course deviate from this as needed.

* Basalt should be added to the Crucible to create Lava
* Basalt should be added to the Electrified Crucible to create Lava

## :anchor: Difficulty
Here is our honest estimate on how difficult (on a scale of 1-5) the implementation may be:

:large_blue_circle::white_circle::white_circle::white_circle::white_circle:

## :construction: Technical Challenges
These are some challenges which may need to be overcome, we wanna be as transparent as possible, so here is some guidance on what may present itself as an obstacle.

*None that we are aware of*

## :memo: Relevant Classes or Snippets
Here are some classes or code snippets which we think might help you get a better understanding where to look for in this gigantic codebase

* [Crucible.java](https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/Crucible.java)
* [ElectrifiedCrucible.java](https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java)

## :book: Useful Resources
If you need help on how to get started, maybe try looking into the following resources!

* Hacktoberfest
    * [Getting started with Hacktoberfest](https://hacktoberfest.digitalocean.com/details#get-started)
    * [Hacktoberfest FAQ](https://hacktoberfest.digitalocean.com/faq)
* GitHub/Open-Source
    * [How to contribute to Open-Source](https://opensource.guide/how-to-contribute/)
    * [Working with forks](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks)
    * [Creating a Pull Request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
* Slimefun
    * [Contributing to Slimefun](https://github.com/Slimefun/Slimefun4/blob/master/CONTRIBUTING.md)
    * [Code of Conduct](https://github.com/Slimefun/Slimefun4/blob/master/.github/CODE_OF_CONDUCT.md)

<hr>

If you want to work on this, simply comment down below and state your interests! Please also comment again if you have changed your mind. Anyone is allowed to discuss in the comments below, feel free to collaborate and work together :heart: <br>
You can always ask for help here if you get stuck.

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description
Added Blackstone and Basalt -> Lava Bucket to crucibles. Duplicate of #2398 I had closed as others had claimed the issue; as discussed in Slimefun's Discord a few days ago, due to inactivity I updated my fork and PR'ed once again.

## Changes
Added Blackstone -> Lava and Basalt -> Lava to Crucible and Electrified Crucible. Following @LinoxGH's suggestion in #2398 I made the recipe cost half as many blocks per Lava Bucket due to materials being Nether related.

## Related Issues
Resolves #2369 

## Checklist
- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
Not applicable:
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java [hunk 1]
```diff
@@ -29,6 +31,13 @@ protected void registerDefaultRecipes() {
         for (Material leaves : Tag.LEAVES.getValues()) {
             registerRecipe(10, new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(leaves, 16) }, new ItemStack[] { new ItemStack(Material.WATER_BUCKET) });
         }
+
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16)) {
+            registerRecipe(10, new ItemStack[] {new ItemStack(Material.BUCKET),
+                new ItemStack(Material.BLACKSTONE, 8)}, new ItemStack[] {new ItemStack(Material.LAVA_BUCKET)});
+            registerRecipe(10, new ItemStack[] {new ItemStack(Material.BUCKET), new ItemStack(Material.BASALT, 8)},
+                new ItemStack[] {new ItemStack(Material.LAVA_BUCKET)});
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:39]: Formatting...
```suggestion
            registerRecipe(10, new ItemStack[] {new ItemStack(Material.BUCKET), new ItemStack(Material.BLACKSTONE, 8)}, new ItemStack[] {new ItemStack(Material.LAVA_BUCKET)});
            registerRecipe(10, new ItemStack[] {new ItemStack(Material.BUCKET), new ItemStack(Material.BASALT, 8)}, new ItemStack[] {new ItemStack(Material.LAVA_BUCKET)});
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by CURVX** [comment]
8 Basalt to Lava is, I feel a bit overpowered. 12, I think will be more balanced since we can make basalt using basalt generator. 8 Blackstone is good.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java (hunk 1 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:39]: I have no clue what happened here, I think copy pasting from GH may not have been the best idea

**Comment by Sfiguz7** [comment]
> 8 Basalt to Lava is, I feel a bit overpowered. 12, I think will be more balanced since we can make basalt using basalt generator. 8 Blackstone is good.

Right, basalt gens. Yea 12 definitely sounds wise

**Comment by LinoxGH** [comment]
I think basalt should give more lava than blackstone or stone or cobble... Anything really. Basalt can be found just near lava, though blackstone is cobblestone but for nether.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/Crucible.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
```
Comment (LinoxGH) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/Crucible.java:10]: Misorganized imports.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines;
 
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
```
Comment (LinoxGH) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:4]: M i s o r g a n i z a t i o n .

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java [hunk 4]
```diff
@@ -29,6 +31,11 @@ protected void registerDefaultRecipes() {
         for (Material leaves : Tag.LEAVES.getValues()) {
             registerRecipe(10, new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(leaves, 16) }, new ItemStack[] { new ItemStack(Material.WATER_BUCKET) });
         }
+
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16)) {
+            registerRecipe(10, new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(Material.BLACKSTONE, 8) }, new ItemStack[] { new ItemStack(Material.LAVA_BUCKET) });
+            registerRecipe(10, new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(Material.BASALT, 12) }, new ItemStack[] { new ItemStack(Material.LAVA_BUCKET) });
```
Comment (LinoxGH) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:37]:
```suggestion
            registerRecipe(10,
                new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(Material.BLACKSTONE, 8) },
                new ItemStack[] { new ItemStack(Material.LAVA_BUCKET) });
            registerRecipe(10,
                new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(Material.BASALT, 12) },
                new ItemStack[] { new ItemStack(Material.LAVA_BUCKET) });
```

**Review by LinoxGH — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java (hunk 4 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:37]: :NotLikeThis:

**Review by TheBusyBiscuit — APPROVED** [review]
Imports could be cleaned up, otherwise: LGTM

**Comment by CURVX** [comment]
> I think basalt should give more lava than blackstone or stone or cobble... Anything really. Basalt can be found just near lava, though blackstone is cobblestone but for nether.

I feel the same. And by this definition, I think anywhere between 2-4 Magma Block to Lava does make sense. Either way, Basalt to Lava is a nice addition :D

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java (hunk 3 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:4]: How are they misorganized?
You wanted them in alphabetical order, they are.

**Comment by Sfiguz7** [comment]
> > I think basalt should give more lava than blackstone or stone or cobble... Anything really. Basalt can be found just near lava, though blackstone is cobblestone but for nether.
> 
> I feel the same. And by this definition, I think anywhere between 2-4 Magma Block to Lava does make sense. Either way, Basalt to Lava is a nice addition :D

Well guys make up your minds. Do we want it to be more logical or more balanced? I don't mind either way but I can't implement both recipes

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java (hunk 3 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectrifiedCrucible.java:4]: They aren't together with the other imports from `io.github.thebusybiscuit.slimefun4.`

**Review by LinoxGH — APPROVED** [review]

**Comment by CURVX** [comment]
> Well guys make up your minds. Do we want it to be more logical or more balanced? I don't mind either way but I can't implement both recipes

I am happy with the nerf changes. 😃👍

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**